checker-framework.yml should be able to fail execution if report generation is failed

detected at https://github.com/checkstyle/checkstyle/actions/runs/7594309984/job/20685718210?pr=14314

According to commit list , https://github.com/checkstyle/checkstyle/commits/master , failure in master started with c403a9a86c2c445d869ee981985344c5eacd9500 which is completely unrelated. Looking at the PR's CI list, the CI did run and said it passed. https://github.com/checkstyle/checkstyle/pull/14311 . The commit before that in master, half the CI ran and checker wasn't one of them.

However, looking at the actual run in https://github.com/checkstyle/checkstyle/pull/14311 you can see that checker throw an error and died. The exception was never reported and made us believe everything was fine.

https://github.com/checkstyle/checkstyle/actions/runs/7593228559/job/20683478761#step:6:470
````
Warning:  Garbage collection consumed 86% of CPU during the past 60 seconds.
Error:  An exception has occurred in the compiler (11.0.21). Please file a bug against the Java compiler via the Java bug reporting page (https://bugreport.java.com/) after checking the Bug Database (https://bugs.java.com/) for duplicates. Include your program and the following diagnostic in your report. Thank you.
com.sun.tools.javac.util.ClientCodeException: java.lang.OutOfMemoryError: Java heap space
	at jdk.compiler/com.sun.tools.javac.api.ClientCodeWrapper$WrappedTaskListener.finished(ClientCodeWrapper.java:832)
	at jdk.compiler/com.sun.tools.javac.api.MultiTaskListener.finished(MultiTaskListener.java:132)
	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1414)
	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1371)
	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:973)
	at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:311)
	at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:170)
	at jdk.compiler/com.sun.tools.javac.Main.compile(Main.java:57)
	at jdk.compiler/com.sun.tools.javac.Main.main(Main.java:43)
Caused by: java.lang.OutOfMemoryError: Java heap space
	at org.checkerframework.framework.source.SourceChecker.wrapThrowableAsBugInCF(SourceChecker.java:2671)
	at org.checkerframework.framework.source.SourceChecker.typeProcess(SourceChecker.java:1059)
	at org.checkerframework.common.basetype.BaseTypeChecker.typeProcess(BaseTypeChecker.java:558)
	at org.checkerframework.javacutil.AbstractTypeProcessor$AttributionTaskListener.finished(AbstractTypeProcessor.java:188)
	at jdk.compiler/com.sun.tools.javac.api.ClientCodeWrapper$WrappedTaskListener.finished(ClientCodeWrapper.java:828)
	... 8 more
````

We need to update the checker CI to detect a OOM exception and mark it as a failure if it occurs.

Edit:
CI runs checker groovy. https://github.com/checkstyle/checkstyle/blob/master/.github/workflows/checker-framework.yml#L54C14-L54C81
Groovy script scans entire maven run, so we should be able to just be able to add a check for the exception. https://github.com/checkstyle/checkstyle/blob/master/.ci/checker-framework.groovy

--- a/.ci/checker-framework.groovy
+++ b/.ci/checker-framework.groovy
@@ -127,7 +127,10 @@ private static List<List<String>> getCheckerFrameworkErrors(final String profile
             checkerFrameworkLines.add(lineFromReader)
             lineFromReader = reader.readLine()
         }
-        process.waitFor()
+        int exitCode = process.waitFor()
+        if (exitCode != 0) {
+            throw new IllegalStateException("Maven process exited with error code: " + exitCode)
+        }
     } finally {
         reader.close()
     }

--- a/pom.xml
+++ b/pom.xml
@@ -2529,7 +2529,7 @@
             <configuration>
               <!-- Must fork or else JVM arguments are ignored. -->
               <fork>true</fork>
-              <failOnError>false</failOnError>
+              <failOnError>true</failOnError>
               <meminitial>1024m</meminitial>
               <maxmem>8192m</maxmem>
               <compilerArguments>
@@ -2588,7 +2588,7 @@
             <configuration>
               <!-- Must fork or else JVM arguments are ignored. -->
               <fork>true</fork>
-              <failOnError>false</failOnError>
+              <failOnError>true</failOnError>
               <meminitial>1024m</meminitial>
               <maxmem>8192m</maxmem>
               <compilerArguments>
@@ -2647,7 +2647,7 @@
             <configuration>
               <!-- Must fork or else JVM arguments are ignored. -->
               <fork>true</fork>
-              <failOnError>false</failOnError>
+              <failOnError>true</failOnError>
               <meminitial>1024m</meminitial>
               <maxmem>8192m</maxmem>
               <compilerArguments>
@@ -2705,7 +2705,7 @@
             <configuration>
               <!-- Must fork or else JVM arguments are ignored. -->
               <fork>true</fork>
-              <failOnError>false</failOnError>
+              <failOnError>true</failOnError>
               <meminitial>1024m</meminitial>
               <maxmem>8192m</maxmem>
               <compilerArguments>
@@ -2762,7 +2762,7 @@
             <configuration>
               <!-- Must fork or else JVM arguments are ignored. -->
               <fork>true</fork>
-              <failOnError>false</failOnError>
+              <failOnError>true</failOnError>
               <meminitial>1024m</meminitial>
               <maxmem>8192m</maxmem>
               <compilerArguments>
@@ -2820,7 +2820,7 @@
             <configuration>
               <!-- Must fork or else JVM arguments are ignored. -->
               <fork>true</fork>
-              <failOnError>false</failOnError>
+              <failOnError>true</failOnError>
               <meminitial>1024m</meminitial>
               <maxmem>8192m</maxmem>
               <compilerArguments>
@@ -2880,7 +2880,7 @@
             <configuration>
               <!-- Must fork or else JVM arguments are ignored. -->
               <fork>true</fork>
-              <failOnError>false</failOnError>
+              <failOnError>true</failOnError>
               <meminitial>1024m</meminitial>
               <maxmem>8192m</maxmem>
               <compilerArguments>
@@ -2939,7 +2939,7 @@
             <configuration>
               <!-- Must fork or else JVM arguments are ignored. -->
               <fork>true</fork>
-              <failOnError>false</failOnError>
+              <failOnError>true</failOnError>
               <meminitial>1024m</meminitial>
               <maxmem>8192m</maxmem>
               <compilerArguments>

